test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs

```diff
@@ -2041,6 +2041,108 @@ public void FilterWithInOperationWithEnums()
             Assert.Equal("FavoriteColors", Assert.IsType<CollectionPropertyAccessNode>(inNode.Right).Property.Name);
         }
 
+        [Fact]
+        public void FilterWithInOperationWithEnumsMemberName_WithoutQualifiedNamespace()
+        {
+            // Arrange
+            string filterQuery = "'SolidYellow' in FavoriteColors";
+
+            string expectedLiteralText = "'SolidYellow'";
+            string expectedfullyQualifiedName = "Fully.Qualified.Namespace.ColorPattern'SolidYellow'";
+            string expectedCollectionPropertyName = "FavoriteColors";
+
+            // Act
+            FilterClause filter = ParseFilter(filterQuery, HardCodedTestModel.TestModel, HardCodedTestModel.GetPersonType());
+
+            // Assert
+            InNode inNode = Assert.IsType<InNode>(filter.Expression);
+            ConstantNode left = Assert.IsType<ConstantNode>(inNode.Left);
+            ODataEnumValue enumValue = Assert.IsType<ODataEnumValue>(left.Value);
+
+            Assert.Equal(expectedLiteralText, left.LiteralText);
+            Assert.Equal(expectedfullyQualifiedName, (enumValue.TypeName + left.LiteralText));
+            Assert.Equal(expectedCollectionPropertyName, Assert.IsType<CollectionPropertyAccessNode>(inNode.Right).Property.Name);
+        }
+
+        [Fact]
+        public void FilterWithInOperationWithEnumsMemberIntegralValue_WithSingleQuotes()
+        {
+            // Arrange
+            string filterQuery = "'12' in FavoriteColors";
+
+            string expectedLiteralText = "'SolidYellow'";
+            string expectedfullyQualifiedName = "Fully.Qualified.Namespace.ColorPattern'SolidYellow'";
+            string expectedCollectionPropertyName = "FavoriteColors";
+
+            // Act
+            FilterClause filter = ParseFilter(filterQuery, HardCodedTestModel.TestModel, HardCodedTestModel.GetPersonType());
+
+            // Assert
+            InNode inNode = Assert.IsType<InNode>(filter.Expression);
+            ConstantNode left = Assert.IsType<ConstantNode>(inNode.Left);
+            ODataEnumValue enumValue = Assert.IsType<ODataEnumValue>(left.Value);
+
+            Assert.Equal(expectedLiteralText, left.LiteralText);
+            Assert.Equal(expectedfullyQualifiedName, (enumValue.TypeName + left.LiteralText));
+            Assert.Equal(expectedCollectionPropertyName, Assert.IsType<CollectionPropertyAccessNode>(inNode.Right).Property.Name);
+        }

```

Comment:
You could use a single parameterized test here, i.e.:

[Theory]
[InlineData("'SolidYellow'")]
[InlineData("'12'")] // Or the equivalent integral value for SolidYellow
public void FilterWithInOperationWithEnumsMemberIntegralValue(string filterOptionValue)
{
    // ...
}